token/js/test/unit/transferHook.test.ts

```diff
@@ -1,173 +1,534 @@
+                // Mocked mint state
+                if (publicKey.equals(mintPubkey)) {
+                    const data = Buffer.alloc(
+                        ACCOUNT_SIZE + ACCOUNT_TYPE_SIZE + TYPE_SIZE + LENGTH_SIZE + TRANSFER_HOOK_SIZE
+                    );
+                    MintLayout.encode(
+                        {
+                            mintAuthorityOption: 0,
+                            mintAuthority: PublicKey.default,
+                            supply: 10000n,
+                            decimals,
+                            isInitialized: true,
+                            freezeAuthorityOption: 0,
+                            freezeAuthority: PublicKey.default,
+                        },
+                        data,
+                        0
+                    );
+                    data.writeUint8(1, ACCOUNT_SIZE); // Account type (1): Mint = 1
+                    data.writeUint16LE(ExtensionType.TransferHook, ACCOUNT_SIZE + ACCOUNT_TYPE_SIZE);
+                    data.writeUint16LE(TRANSFER_HOOK_SIZE, ACCOUNT_SIZE + ACCOUNT_TYPE_SIZE + TYPE_SIZE);
+                    TransferHookLayout.encode(
+                        {
+                            authority: Keypair.generate().publicKey,
+                            programId: transferHookProgramId,
+                        },
+                        data,
+                        ACCOUNT_SIZE + ACCOUNT_TYPE_SIZE + TYPE_SIZE + LENGTH_SIZE
+                    );
+                    return {
+                        data,
+                        owner: TOKEN_2022_PROGRAM_ID,
+                        executable: false,
+                        lamports: 0,
+                    };
+                }
+
+                // prettier-ignore
+                // Mocked validate state
+                if (publicKey.equals(validateStatePubkey)) {
+                const extraAccountsList: ExtraAccountMetaList =  {
+                    count: extraAccounts.length,
+                    extraAccounts,
+                }
+                const instructionDiscriminator = Buffer.from([105, 37, 101, 197, 75, 251, 102, 26]).readBigUInt64LE();
+                const data = Buffer.alloc(8 + 4 + 4 + ExtraAccountMetaLayout.span * extraAccounts.length);
+                ExtraAccountMetaAccountDataLayout.encode({
+                    instructionDiscriminator,
+                    length: 4 + ExtraAccountMetaLayout.span * extraAccounts.length,
+                    extraAccountsList,
+                }, data);
+                return {
+                    data,
+                    owner: transferHookProgramId,
+                    executable: false,
+                    lamports: 0,
+                };
+            }
+
+                return {
+                    data: Buffer.from([]),
+                    owner: PublicKey.default,
+                    executable: false,
+                    lamports: 0,
+                };
+            };
```

Comment:
I'm not sure if it's good to use prettier-ignore here, since it made this part harder to read. Is there any way to fixup the spacing or object declarations so that it plays well with prettier?

Reply:
I can take it out there. I really only wanted it where the raw bytes are. This is what it looks like if I take it out:

```ts
if (publicKey.equals(validateStatePubkey)) {
    const extraAccountsList: ExtraAccountMetaList = {
        count: extraAccounts.length,
        extraAccounts,
    };
    const instructionDiscriminator = Buffer.from([
        105, 37, 101, 197, 75, 251, 102, 26,
    ]).readBigUInt64LE();
    const data = Buffer.alloc(8 + 4 + 4 + ExtraAccountMetaLayout.span * extraAccounts.length);
    ExtraAccountMetaAccountDataLayout.encode(
        {
            instructionDiscriminator,
            length: 4 + ExtraAccountMetaLayout.span * extraAccounts.length,
            extraAccountsList,
        },
        data
    );
    return {
        data,
        owner: transferHookProgramId,
        executable: false,
        lamports: 0,
    };
}
```

Reply:
It's *a little* easier to read, ha 😅, wdyt?

Reply:
Yeah that's better!